lib/puppet/parser/functions/ip_in_range.rb

```diff
@@ -0,0 +1,57 @@
+module Puppet::Parser::Functions
+  newfunction(:ip_in_range, :type => :rvalue, :arity => 2, :doc => <<-DOC
+    ip_in_range returns true if the ipaddress is within the given CIDRs
+    usage: ip_in_range(<IPv4 Address>, <IPv4 CIDR>)
+
+    $valid_ip = ip_in_range('10.10.10.53', '10.10.10.0/24') # $valid_ip == true
+
+    or you can pass an array of CIDR ranges and return true if the IP is inside
+    any of them.
+
+    $ranges = ['192.168.0.0/24', '10.10.10.0/24']
+    $valid_ip = ip_in_range('10.10.10.53', $ranges) # $valid_ip == true
+
+    # or use it in a conditional
+
+    if ip_in_range($ipaddress, '10.10.200.0/24') == true {
+      notify { 'Do subnet based resources here': }
+    }
+    DOC
+             ) do |arguments|
+
+    require 'ipaddr'
+
+    unless arguments.length == 2
+      raise ArgumentError, "ip_in_range(): wrong number of arguments (#{arguments.length} must be 2)"
+    end
+
+    # load in the ip validation function from puppetlabs-stdlib
+    unless Puppet::Parser::Functions.function('is_ip_address')
+      raise Puppet::ParseError, 'ip_in_range(): requires the puppetlabs-stdlib functions'
+    end
+
+    # 2nd arg can be an array so normalise it into one array for processing
+    arguments.flatten!
+
+    # error out if any of the arguments given are not valid IP addresses
+    arguments.each do |ipaddress|
+      unless function_is_ip_address([ipaddress])
```

Comment:
This is old API that does not work with modern functions - use `call_function(name, ...)` instead as that can call both 3x and 4x functions.

Reply:
Also, do you actually need to typecheck the input first since you would be getting an error from `IPAddr.new`